File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example_py/h10-w/params/get_tcp_offset.py
import humanoid_sdk_py
import humanoid_sdk_py.h10w as h10w
import time

h10w_params = h10w.H10wParams()
time.sleep(1)

print("=== 获取 TCP 偏移量示例 ===")

ret, tcp_offset_params = h10w_params.getTcpOffset()
if ret == 0:
    print("获取成功")
    for index, tcp_offset in tcp_offset_params:
        print("    TCP {}: {}".format(index, tcp_offset))
else:
    print("获取失败")

File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example/h10-w/params/get_tcp_offset.cpp
// get_tcp_offset.cpp

#include <iostream>
#include <iterator>
#include <algorithm>
#include <iomanip>
#include "h10-w/h10w_params.h"
#include "common/common_def.h"

using namespace humanoid;
using namespace humanoid::h10w;

int main() {
  std::cout << "=== 获取 TCP 偏移量示例 ===" << std::endl;
  H10wParams params;
  TcpOffsetParams tcpOffsetParams = {
      // 留空则返回全部 TCP 偏移量
      {1, {}},  // 获取左臂 TCP 偏移量
      {2, {}}   // 获取右臂 TCP 偏移量
  };

  if (!params.getTcpOffset(tcpOffsetParams)) {
    std::cout << "获取成功" << std::endl;
    for (const auto& [index, offset] : tcpOffsetParams) {
      std::cout << "    TCP " << index << ": ";
      std::copy(offset.begin(), offset.end(),
                std::ostream_iterator<double>(std::cout, " "));
      std::cout << std::endl;
    }
  } else {
    std::cout << "获取失败" << std::endl;
    return -1;
  }
  return 0;
}
